### PR TITLE
Fix battery handling

### PR DESCRIPTION
**Pull request by btry**
see 

incomplete spec  for JSON format
https://github.com/glpi-project/inventory_format/pull/18

XML spec 
https://github.com/fusioninventory/fusioninventory.github.io/pull/75

Bad value collected by the agent
https://github.com/fusioninventory/fusioninventory-agent/issues/840
https://github.com/fusioninventory/fusioninventory-agent/pull/843

support for current max capacity to monitor battery wear
https://github.com/glpi-project/glpi/pull/8062

### CODE REVIEW

File: inc/inventorycomputerlib.class.php [hunk 1]
```diff
@@ -1342,28 +1342,11 @@ function updateComputer($a_computerinventory, $computers_id, $no_history, $setdy
...
-                     }
-
-                     if ($arraydb['voltage'] == 0
-                              && $arrayslower['voltage'] > 0) {
-                        $input = [
-                           'id'        => $keydb,
-                           'voltage'   => $arrayslower['voltage']
-                        ];
-                        $update = true;
-                     }
```
Comment (btry) [inline inc/inventorycomputerlib.class.php:1362]: Voltage belongs to DeviceBattery, not Item_DeviceBattery. It is useless to handle it here as the column does not exists.

File: inc/inventorycomputerlib.class.php [hunk 2]
```diff
@@ -1342,28 +1342,11 @@ function updateComputer($a_computerinventory, $computers_id, $no_history, $setdy
...
-
-                     if ($update === true) {
-                        $item_DeviceMemory->update($input);
-                     }
+                     $input = [
+                        'id' => $keydb,
+                        'real_capacity' => $arrayslower['real_capacity'],
+                     ];
```
Comment (btry) [inline inc/inventorycomputerlib.class.php:1348]: Support for real_capacity, which beslongs to Item_DeviceBttery

File: inc/inventorycomputerlib.class.php [hunk 3]
```diff
@@ -1342,28 +1342,11 @@ function updateComputer($a_computerinventory, $computers_id, $no_history, $setdy
...
-                              && $arrayslower['voltage'] > 0) {
-                        $input = [
-                           'id'        => $keydb,
-                           'voltage'   => $arrayslower['voltage']
-                        ];
-                        $update = true;
-                     }
-
-                     if ($update === true) {
-                        $item_DeviceMemory->update($input);
```
Comment (btry) [inline inc/inventorycomputerlib.class.php:1365]: bad object here

**Comment by trasher** [comment]
Please fix tests

**Review by trasher — APPROVED** [review]
Sounds good to me, even if not tested (waiting for next agent release)

**Review by cedric-anne — APPROVED** [review]

**Comment by btry** [comment]
@trasher : To test the latest agent you can clone the repo and run 

```
perl bin/fusioninventory-agent -s https://server/glpi/plugins/fusioninventory
```

or 
```
perl bin/fusioninventory-agent -l /tmp/inventory.xml
````
then import the xml in Fusion Inventory.

**Comment by trasher** [comment]
Code assumes capacity (and voltage) is an integer, but in my case, this is not:
```
    <BATTERIES>
      <CAPACITY>29.07 Wh</CAPACITY>
      <CHEMISTRY>lithium-polymer</CHEMISTRY>
      <DATE>10/11/2015</DATE>
      <MANUFACTURER>SMP</MANUFACTURER>
      <NAME>DELL JHXPY53</NAME>
      <SERIAL>3525</SERIAL>
      <VOLTAGE>8.497 V</VOLTAGE>
    </BATTERIES>
```

Result in glpi:
image

**Comment by trasher** [comment]
(using fusion agent 2.5.2)

**Comment by g-bougard** [comment]
@trasher If you're still using agent 2.5.2, this XML part is simply wrong. So not the plugin fault, but the agent one. Using agent 2.6 will fix this problem.

**Comment by trasher** [comment]
Agent data is not wrong, but not the correct unit

**Comment by g-bougard** [comment]
Indeed the agent did not respect the defined specs and the specs tell to provide an integer normalized to mWh for CAPACITY and mV for VOLTAGE. So sorry, but in your case, the agent was really wrong as it should have normalize values and only send the raw values without unit.

**Comment by trasher** [comment]
All is fine then

**Review by g-bougard — APPROVED** [review]

**Pull request merged**